src/address_space.rs

```diff
@@ -124,13 +139,25 @@ impl AddressSpace {
         &self,
         addr: VirtualAddress,
         access_type: FlagBuilder,
-    ) -> Result<(Arc<D>, usize), &str> {
-        todo!();
+    ) -> Result<(Arc<dyn DataSource>, usize), &str> {
+        if addr > VADDR_MAX {
```

Comment:
If the address is greater than the max address, how would you ever have a mapping there?